pkg/services/searchV2/auth.go

```diff
@@ -6,22 +6,22 @@ import (
 	"github.com/grafana/grafana/pkg/infra/db"
 	"github.com/grafana/grafana/pkg/models"
 	"github.com/grafana/grafana/pkg/services/accesscontrol"
+	"github.com/grafana/grafana/pkg/services/dashboards"
 	"github.com/grafana/grafana/pkg/services/sqlstore/permissions"
-	"github.com/grafana/grafana/pkg/services/sqlstore/searchstore"
 	"github.com/grafana/grafana/pkg/services/user"
 )
 
 // ResourceFilter checks if a given a uid (resource identifier) check if we have the requested permission
-type ResourceFilter func(uid string) bool
+type ResourceFilter func(kind entityKind, uid, parent string) bool
 
 // FutureAuthService eventually implemented by the security service
 type FutureAuthService interface {
 	GetDashboardReadFilter(user *user.SignedInUser) (ResourceFilter, error)
 }
 
-var _ FutureAuthService = (*simpleSQLAuthService)(nil)
+var _ FutureAuthService = (*simpleAuthService)(nil)
 
-type simpleSQLAuthService struct {
+type simpleAuthService struct {
```

Comment:
nit/post merge: can we extended `Auth` to `Authorization` in the names here and rename the auth.go to authz.go after?

Reply:
Yup, sounds good 👍